### PR TITLE
DOC: add example & prose of slicing with labels when index has duplicate labels 

### LINKED ISSUE #36251: DOC: slicing with labels when index has duplicate labels
#### Location of the documentation

https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#slicing-ranges

**Note**: You can check the latest versions of the docs on `master` [here](https://pandas.pydata.org/docs/dev/).

#### Documentation problem

Documentation doesn't specify what happens when you slice an index that has duplicates.  Does it denote the range from the first occurrence of the start element, to the last occurrence of the end element?

#### Suggested fix for documentation

Clarify how slicing interacts with non-unique index elements.

I would also emphasize, at 
https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#slicing-with-labels
that the actual values of the index labels, and their natural comparison order and the elements between them in that natural order, do _not_ matter for slicing; i.e. that a slice of 'Z':'A' is not empty but depends on the positions that the labels denote in the array.   In all current documentation examples, the label denoting the start of the slice end also happens to be lexicographically smaller than the label denoting the end of the slice, which can mislead readers into thinking the natural order of the labels has some meaning, when all they do is denote integer positions.

### PR DESCRIPTION
**Pull request by junjunjunk**
- [x] closes #36251

### CODE REVIEW

**Comment by junjunjunk** [comment]
Can someone please review this?
would love to get a hacktoberfest tag if possible. #36837

File: doc/source/user_guide/indexing.rst [hunk 1]
```diff
@@ -422,6 +422,17 @@ above example, ``s.loc[1:6]`` would raise ``KeyError``.
 For the rationale behind this behavior, see
 :ref:`Endpoints are inclusive <advanced.endpoints_are_inclusive>`.
 
+.. ipython:: python
+
+   s = pd.Series(list('abcdef'), index=[0, 3, 2, 5, 4, 2])
+   s.loc[3:5]
+
+Also, if the index has duplicate labels *and* either the start or the stop labels is dupulicate labels,
```
Comment (dsaxton) [inline doc/source/user_guide/indexing.rst:430]:
```suggestion
Also, if the index has duplicate labels *and* either the start or the stop label is duplicated,
```

File: doc/source/user_guide/indexing.rst [hunk 2]
```diff
@@ -422,6 +422,17 @@ above example, ``s.loc[1:6]`` would raise ``KeyError``.
...
+an error will be raised. For instance, in the above example, ``s.loc[2:5]`` would raise ``KeyError``.
```
Comment (dsaxton) [inline doc/source/user_guide/indexing.rst:431]:
```suggestion
an error will be raised. For instance, in the above example, ``s.loc[2:5]`` would raise a ``KeyError``.
```

File: doc/source/user_guide/indexing.rst (hunk 1 above)
Comment (junjunjunk) [reply doc/source/user_guide/indexing.rst]: Thanks! Done.

File: doc/source/user_guide/indexing.rst (hunk 2 above)
Comment (junjunjunk) [reply doc/source/user_guide/indexing.rst]: Done.

**Review by dsaxton — APPROVED** [review]
Thanks @junjunjunk

**Comment by junjunjunk** [comment]
Thanks! @dsaxton

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @junjunjunk. Sorry for letting this sit for so long.

**Pull request merged**